lib/granite/action/transaction.rb

```diff
@@ -6,34 +8,27 @@ class Rollback < defined?(ActiveRecord) ? ActiveRecord::Rollback : StandardError
     module Transaction
       extend ActiveSupport::Concern
 
-      private
+      included do
+        define_callbacks :commit
+      end
+
+      module ClassMethods
+        delegate :transaction, to: :'Granite::Action::TransactionManager'
 
-      def transactional(&block)
-        if transactional?
-          yield
-        else
-          @_transactional = true
-          result = transaction(&block) || false
-          @_transactional = nil
-          result
+        private
+
+        # Defines a callback which will be triggered right after transaction committed
+        # Uses the same arguments as `ActiveSupport::Callbacks.set_callback`
```

Comment:
Except for the first two arguments actually, so maybe it makes sense to clarify that or copy it over here.